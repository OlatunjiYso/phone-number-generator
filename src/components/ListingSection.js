import React from 'react';


import NumberAnalytics from './NumberAnalytics';

function ListingSection(props) {
  const { existingNumbers, lowestNumber, highestNumber, sortInAscending, sortInDescending } = props;
  const displayedNumbers = existingNumbers.map((number, index) => (
    <h3 key={index} className="phone-number"> {number} </h3>
  ))
  return (
    <div className="listing-section">
      <div className="number-analytics">
        <NumberAnalytics 
        existingNumbers={existingNumbers}
        lowestNumber={lowestNumber}
        highestNumber={highestNumber}
        sortInAscending={sortInAscending}
        sortInDescending={sortInDescending}
        />
      </div>
      <div >
        {displayedNumbers}
      </div>
    </div>
  );
}

export default ListingSection;
