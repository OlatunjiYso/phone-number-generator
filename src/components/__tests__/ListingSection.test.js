import React from 'react';
import { shallow } from 'enzyme';
import ListingSection from '../ListingSection';

it('renders the ListingSection component', () => {
  const wrapper = shallow(<ListingSection existingNumbers={[]}/>);
  expect(wrapper).toMatchSnapshot();
});