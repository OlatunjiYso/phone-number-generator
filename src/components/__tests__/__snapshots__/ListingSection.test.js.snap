// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ListingSection component 1`] = `ShallowWrapper {}`;

exports[`renders the ListingSection component 2`] = `ShallowWrapper {}`;
